src/selectors/per-thread/markers.js

```diff
@@ -94,19 +97,53 @@ export function getMarkerSelectorsPerThread(threadSelectors: *) {
       )
   );
 
-  const getCommittedRangeFilteredMarkers: Selector<Marker[]> = createSelector(
-    getReferenceMarkerTable,
+  const getMarkerGetter: Selector<(MarkerIndex) => Marker> = createSelector(
+    getFullMarkerList,
+    markerList => (markerIndex: MarkerIndex): Marker => {
+      const marker = markerList[markerIndex];
+      if (!marker) {
+        throw new Error(stripIndent`
+          Tried to get marker index ${markerIndex} but it's not in the full list.
+          This is a programming error.
+        `);
+      }
+      return marker;
+    }
+  );
+
+  const getReferenceMarkerIndexes: Selector<MarkerIndex[]> = createSelector(
```

Comment:
What do you think about `getFullMarkerListIndexes` instead, to avoid introducing another name?

Reply:
oh, totally, I forgot to rename it!